### DISCUSSION TITLE
TSML Plugin Version 3.12.1

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#602] [General]
I see there is an update to the TSML plugin to Version 3.12.1.

The details tell me this version has the following, but I'd like some clarification on some of the changes. My specific question is in italics.

- Add CSS class for past meetings _- please describe what this is_
- Allow translations of attendance options _- is this for different languages?_
- Use a default meeting title if left blank 
- Add CMA support
- Create feature_request.yml issue template _- what is this?_

Thanks

**Comment by tech2serve** [comment]
The CSS class was added to the allow webmasters to specify styling so the website will show past meetings differently. This could be greyed out, italicized, etc. @brianw-area24 or @97eighty can provide more information.

Yes, the translations of attendance options is for French. English sites don’t need to worry about it.

In hindsight, we shouldn’t have included mention of the feature_request.yml as this is really for developers. It doesn’t affect the plugin itself at all.

**Reply by Grateful-Viv** [reply]
Thanks for quick reply. By past meetings, do you mean suspended meetings? I’m looking forward to hearing from @brianw-area24 and/or @97eighty for further elaboration.

**Reply by tech2serve** [reply]
No. We’ve been adopting the requested feature of haven’t meetings that are already in progress show up in the widget. For example, say it is 10:11a. In the past, meetings that started at 10a wouldn’t show. Then we started them showing, but this change goes a step further and allows the admin to style those differently than future meetings.

**Comment by 97eighty** [comment]
@Grateful-Viv When filtering meetings by day you'll notice that the meetings which have already occurred display with the color "red". We color these meetings "red" by default, using:

```
.past {
    color: red;
}
```
Using a CSS class to define the color allows users to override the class in their own CSS:

```
.past {
    color: #ffa500; // A shade of orange
}
```
Depending on where this color change occurs a user may have to add the !important rule.

**Reply by Grateful-Viv** [reply]
Thank you. 

I have this installed on my staging site and saw that none of the previous meetings were shaded by default. When I added the CSS as shown in your example, it did change the color for all previous meetings today. I was hoping it would color the meetings in progress that may have started, say 15 minutes ago, which is how I've seen TSMl_ui appear.

Also, when I filtered to meetings any day (/meetings/?tsml-day=any) and then back to today (/meetings/?tsml-day=4), the colors did not change in accordance with the CSS. That's with or without !important;

Below are the screenshots of the plugin and the TSML-ui for the same meetings. 

I like the way the ui works in that it shows 3 meetings in progress with a brown bar and if I select that, it displays the meetings currently in progress.

![plugin meetings in progress](https://user-images.githubusercontent.com/62247421/147305086-88f8d25c-2ede-4165-af99-640fb0f5ad17.png)

The plugin just shows all meetings that have expired today. That is not very helpful. And nothing is shown on the upcoming meeting widget.

![meetings in progress](https://user-images.githubusercontent.com/62247421/147305101-e9b87d14-5110-418f-beec-81e6b007164f.png)

**Reply by Grateful-Viv** [reply]
I'd also like to add that I would use the ui, which I like a lot better, but the widgets don't work with this. I specifically use the upcoming widget in our sidebar on most pages (not the meeting page), as well as the Meetings Top and the Meetings Detail Bottom. 

The Update Meeting Info is okay and I could work with that, or find a way to incorporate my own form.

I do use the plugin front-end to edit, rather than either using the backend, since it's much faster to add or update meetings. I usually do not upload and replace the csv files. We don't have that many. At some point, I am not going to be the trusted servant for maintaining this, so I need a way to make this as painless and intuitive as possible for the next person.

**Reply by Grateful-Viv** [reply]
Can someone respond to this please? It may have gotten buried over the holiday. Thanks so much.

> I have this installed on my staging site and saw that none of the previous meetings were shaded by default. When I added the CSS as shown in your example, it did change the color for all previous meetings today. I was hoping it would color the meetings in progress that may have started, say 15 minutes ago, which is how I've seen TSMl_ui appear.
> 
> Also, when I filtered to meetings any day (/meetings/?tsml-day=any) and then back to today (/meetings/?tsml-day=4), the colors did not change in accordance with the CSS. That's with or without !important;
> 
> Below are the screenshots of the plugin and the TSML-ui for the same meetings.
> 
> I like the way the ui works in that it shows 3 meetings in progress with a brown bar and if I select that, it displays the meetings currently in progress.
> 
> ![plugin meetings in progress](https://user-images.githubusercontent.com/62247421/147305086-88f8d25c-2ede-4165-af99-640fb0f5ad17.png)
> 
> The plugin just shows all meetings that have expired today. That is not very helpful. And nothing is shown on the upcoming meeting widget.
> 
> ![meetings in progress](https://user-images.githubusercontent.com/62247421/147305101-e9b87d14-5110-418f-beec-81e6b007164f.png)

**Reply by 97eighty** [reply]
@Grateful-Viv Currently, the .past class is applied when the meetings are first displayed and not applied when the filters are being used. I'm not sure that was the intended use of the class though. 

In the near future we'll be integrating the TSML UI more closely in the WordPress product and I'm sure we'll include the Before and After widgets as well. If that is holding you back from using TSML UI then will be included with the UI. 

If your desire is to use the new TSML UI and still use the Widgets it can be solved with code or a template. These would require  a programmer. You could explore using some custom code:

https://gist.github.com/97eighty/fd15dff3f471db2dff5d76a3d22f6821

And then use this code like:

[widget widget_name="tsml_meeting_top"]
[tsml_ui]
[widget widget_name="tsml_meeting_bottom"]

**Reply by 97eighty** [reply]
Having those Widgets available as shortcodes is worth us discussing in an upcoming release.

**Comment by 97eighty** [comment]
@Grateful-Viv And, the YML files are specially formatted forms used inside of GITHUB to collect specific information about bug reports and feature requests. They are intended for developers using GITHUB.
